### PR TITLE
Don't process data for stack pages

### LINKED ISSUE #1204: Don't retain data for stack pages
### File Location(s)

core-processor/src/executor.rs#L157-L166

### Proposal

The stack pages aren't used after execution so there is no reason to process its data.

### PR DESCRIPTION
**Pull request by gshep**
Resolves #1204 

@gear-tech/dev

### CODE REVIEW

File: core-processor/src/ext.rs [hunk 1]
```diff
@@ -233,6 +234,7 @@ impl IntoExtInfo for Ext {
         let mut pages_data = BTreeMap::new();
         for page in (0..static_pages.0)
             .map(WasmPageNumber)
+            .skip_while(|page| *page < stack_page_count)
```
Comment (grishasobol) [inline core-processor/src/ext.rs:237]: May be it's better to use `stack_page_count.0..static_pages.0`. Check before that stack_page_count <= static_pages

**Review by grishasobol — APPROVED** [review]

File: core-processor/src/ext.rs (hunk 1 above)
Comment (gshep) [reply core-processor/src/ext.rs:237]: Is `static_page_count > static_pages` case possible?

Comment (grishasobol) [reply core-processor/src/ext.rs:237]: > static_page_count > static_pages case possible?
Yes it's possible if contract has wrong information in __gear_end_stack export.

I don't mind

File: core-backend/sandbox/src/env.rs [hunk 2]
```diff
@@ -232,21 +232,25 @@ where
...
-        let (info, trap_explanation) =
-            ext.into_inner()
-                .into_ext_info(&memory)
-                .map_err(|(reason, gas_amount)| BackendError {
-                    reason: SandboxEnvironmentError::Memory(reason),
-                    gas_amount,
-                })?;
+        let (info, trap_explanation) = ext
+            .into_inner()
+            .into_ext_info(&memory, stack_end_page.unwrap_or_default())
```
Comment (breathx) [inline core-backend/sandbox/src/env.rs:249]: isn't absence of stack end a critical error for us?

**Review by breathx — APPROVED** [review]

File: core-backend/sandbox/src/env.rs (hunk 2 above)
Comment (gshep) [reply core-backend/sandbox/src/env.rs:249]: Nope, since codegenerator doesn't create `__stack_end` for every program. I guess if it small enough.

**Pull request merged**